packages/styled-components/src/utils/flatten.js

```diff
@@ -14,31 +14,34 @@ import addUnitIfNeeded from './addUnitIfNeeded';
  */
 const isFalsish = chunk => chunk === undefined || chunk === null || chunk === false || chunk === '';
 
-export const objToCssArray = (obj: Object, prevKey?: string): Array<string | Function> => {
-  const rules = []
-  const keys = Object.keys(obj)
-
-  keys
-    .forEach(key => {
-      if (!isFalsish(obj[key])) {
-        if (isPlainObject(obj[key])) {
-          rules.push(...objToCssArray(obj[key], key))
-
-          return rules
-        }
-        else if (isFunction(obj[key])) {
-          rules.push(`${hyphenate(key)}:`, obj[key], ';')
-
-          return rules
-        }
-        rules.push(`${hyphenate(key)}: ${addUnitIfNeeded(key, obj[key])};`);
-      }
-      return rules;
-    })
-
-    return prevKey
-      ? [`${prevKey} {`, ...rules, '}']
-      : rules;
+/**
+ * Reduces over the object converting into a template string args array.
+ */
+export const objToCssArray = (
+  obj: Object,
+  styleSheet: ?Object,
+  executionContext: ?Object,
+  prevKey?: string
+): Array<string | Function> => {
+  const rules = Object.keys(obj).reduce((acc, key) => {
+    const val = obj[key];
+    if (isFalsish(val)) return acc;
+
+    if (isPlainObject(val)) {
+      return acc.concat(...objToCssArray(val, styleSheet, executionContext, key));
```

Comment:
Looking at this again, this spread is redundant (copied from the previous version of the code) because the `concat` will have the same effect. I'll remove it.